ast/zson.go

```diff
@@ -0,0 +1,174 @@
+package ast
+
+type Value interface {
+	valueNode()
+}
+
+const (
+	ImpliedValueOp = "implied_value"
+	CastValueOp    = "cast_value"
+	DefValueOp     = "def_value"
```

Comment:
Nit: We usually (but admittedly not always) have a common prefix on constants like these and those below (i.e., OpImpliedValue), and I favor that scheme.

Reply:
All of the ast nodes in package ast should have *Op constants so let's hold off on this rename until we do that.  There is code all over the puts a string in for the Op field and somethimes it's left foo so I like the pattern ...
```
return &Thing{
  Op: ThingOp,
  Rest: ...
}
```
We can bikeshed whether its Op* or *Op when we get to this :-)
p.s. Op here doesn't mean operator but rather Opcode for the JSON object type that is being coded.  When we switch this to use ZSON instead of JSON encoding, we can elimiate Op everywhere and use type inference.  But we can't do this until the (javascript) client can do a ZSON.Parse() instead of JSON.Parse() to recover what will have been previously encoded in the "op" key of the JSON encoding.